### PR TITLE
4860-deprecate-and-remove-dashboards-code

### LINKED ISSUE #4860: Deprecate and remove "dashboards" code
None of this provides any real value. Adding a custom dashboard is trivial and I really don't think we need a custom permission set and empty controller/view. 

- https://github.com/solidusio/solidus/blob/master/backend/app/controllers/spree/admin/dashboards_controller.rb
- https://github.com/solidusio/solidus/blob/master/backend/app/views/spree/admin/dashboards/home.html.erb
- https://github.com/solidusio/solidus/blob/master/core/lib/spree/permission_sets/dashboard_display.rb
- https://github.com/solidusio/solidus/blob/master/core/spec/models/spree/permission_sets/dashboard_display_spec.rb

### PR DESCRIPTION
**Pull request by lauriejefferson**
## Summary

  Fixes #4860 
  This PR adds deprecation warnings to the following views and controllers:

-   [DashboardsController](https://github.com/solidusio/solidus/blob/main/backend/app/controllers/spree/admin/dashboards_controller.rb)
- [Home View](https://github.com/solidusio/solidus/blob/main/backend/app/views/spree/admin/dashboards/home.html.erb)
- [Dashboards Display](https://github.com/solidusio/solidus/blob/main/core/lib/spree/permission_sets/dashboard_display.rb)
- [Dashboards Display Spec](https://github.com/solidusio/solidus/blob/main/core/spec/models/spree/permission_sets/dashboard_display_spec.rb)

### CODE REVIEW

**Comment by jarednorman** [comment]
Thanks for using the deprecator. This still hasn't addressed my other concerns:

> Additionally, the DashboardsController is deprecated in a way that simply loading it will trigger the deprecation which is undesirable and the specs themselves don't need to be deprecated. I recommend looking through some past PRs that deprecated things to find some examples of how we've handled similar deprecations in the past to get a feel for how this should be handled.

**Comment by lauriejefferson** [comment]
I need a little help with the DashboardController deprecation.  I searched through past PRs and I see #4102 that may relate.  From the PR, It does use a deprecator in the OrdersController, but I'm not sure if that code has also been deprecated in favor of Spree.deprecator.

File: backend/app/controllers/spree/admin/dashboards_controller.rb
Comment (tvdeyen) [inline backend/app/controllers/spree/admin/dashboards_controller.rb:8]: I think if we put this in a `before_action` the code would only run if the controller is used during runtime and not during app boot. Could you try?

```rb
before_action do
  Spree.deprecator.warn "The Dashboards controller is deprecated. Please use the new admin dashboard."
end
```

Comment (tvdeyen) [reply backend/app/controllers/spree/admin/dashboards_controller.rb:8]: Also I find the message a bit confusing. To me it is not clear which one is deprecated and which I should use.

The deprecated controller can simply be logged by using

```rb
"The #{self.class.name} controller" ...
```

in the message. Could you please give a more precise hint which the new controller is? Maybe by using a path or a more precise name?

Comment (lauriejefferson) [reply backend/app/controllers/spree/admin/dashboards_controller.rb:8]: Updated controller.

Comment (jarednorman) [reply backend/app/controllers/spree/admin/dashboards_controller.rb:8]: There isn't a new admin dashboard. This dashboards feature was something that was added to the project, but never actually used in core (and I've never seen it used in store either.) We're not removing this in place of something else, we're just removing it.

Comment (kennyadsl) [reply backend/app/controllers/spree/admin/dashboards_controller.rb:8]: Yes, for now there's no replacement. We can say something like:

"The Dashboards controller is deprecated. If you still use dashboards, please copy all controllers and views from solidus_backend to your application."

Comment (kennyadsl) [reply backend/app/controllers/spree/admin/dashboards_controller.rb:8]: Also, there's a test failure that I think is related!

**Review by kennyadsl — COMMENTED** [review]
Thanks for the contribution @lauriejefferson! I left a couple of comments, but I think we are close. 🙂

File: backend/app/controllers/spree/admin/dashboards_controller.rb
Comment (lauriejefferson) [reply backend/app/controllers/spree/admin/dashboards_controller.rb:8]: @kennyadsl Is the above language the same for the DashboardDisplay module, the Home controller and view?

Comment (kennyadsl) [reply backend/app/controllers/spree/admin/dashboards_controller.rb:8]: Maybe we can use a single message that is more generic and we can apply to all. Up to you to choose it, but let me know if I can help.

Comment (lauriejefferson) [reply backend/app/controllers/spree/admin/dashboards_controller.rb:9]: See latest commit.

**Comment by lauriejefferson** [comment]
@kennyadsl @jarednorman Can this PR be closed?

**Review by jarednorman — APPROVED** [review]
Yeah, I think this looks good now.

**Comment by tvdeyen** [comment]
@lauriejefferson would you mind to rebase and force push your branch with latest `main` to retrigger the pipepline? Thanks 🙏🏻

**Comment by lauriejefferson** [comment]
@tvdeyen Rebase completed.

**Comment by lauriejefferson** [comment]
@jarednorman @tvdeyen @kennyadsl Is there any feedback on this issue?

**Comment by jarednorman** [comment]
Nope, I have approved and am just waiting for another person on core to approve.

**Comment by tvdeyen** [comment]
@lauriejefferson the spec failures are related. Mind taking another look?

File: core/spec/models/spree/permission_sets/dashboard_display_spec.rb
Comment (tvdeyen) [inline core/spec/models/spree/permission_sets/dashboard_display_spec.rb:13]: Ah, a typo causes the spec error

```suggestion
      Spree.deprecator.warn 'The dashboard_display_spec.rb test is being deprecated.  Please update your source code.'
```

File: core/spec/models/spree/permission_sets/dashboard_display_spec.rb
Comment (tvdeyen) [inline core/spec/models/spree/permission_sets/dashboard_display_spec.rb:22]:
```suggestion
    Spree.deprecator.warn 'The dashboard_display_spec.rb test is being deprecated.  Please update your source code.'
```

**Comment by lauriejefferson** [comment]
@tvdeyen I fixed the rspec tests.  Please review.

**Review by jarednorman — APPROVED** [review]

**Comment by nvandoorn** [comment]
Hey @lauriejefferson, I have a revised version of this branch available here: https://github.com/nvandoorn/solidus/tree/4860-deprecate-and-remove-dashboards-code

The specs should pass and everything is up to date. You're welcome to pull these changes if you wish, or I can open another pull request.

**Comment by lauriejefferson** [comment]
@nvandoorn Has your branch been approved? I'm not sure if I need to pull your changes.  I may have added what was necessary to complete this issue. @jarednorman @tvdeyen Please advise.

**Comment by nvandoorn** [comment]
Hey @lauriejefferson, I haven't opened a pull request yet so the changes have not been approved yet.

I believe the current revision of this pull request needs a revision  for the specs to pass as indicated by @tvdeyen's last comment.

**Comment by lauriejefferson** [comment]
@jarednorman @tvdeyen @nvandoorn I pulled @nvandoorn changes into my local branch and everything's up to date.  Can you verify that the correct specs have been added?  Please advise.

**Comment by jarednorman** [comment]
I think we're good.

**Comment by lauriejefferson** [comment]
@jarednorman @kennyadsl Can this PR be closed?

**Comment by kennyadsl** [comment]
@lauriejefferson thanks for the contribution!

**Pull request closed without merging**
